relayer/chains/wasm/provider.go

```diff
@@ -514,11 +514,11 @@ func (p *Provider) getHeightStream(done <-chan bool, fromHeight, toHeight uint64
 	heightChan := make(chan *types.HeightRange)
 	go func(fromHeight, toHeight uint64, heightChan chan *types.HeightRange) {
 		defer close(heightChan)
-		for fromHeight < toHeight {
+		for fromHeight <= toHeight {
 			select {
 			case <-done:
 				return
-			case heightChan <- &types.HeightRange{Start: fromHeight, End: fromHeight + p.cfg.BlockBatchSize}:
+			case heightChan <- &types.HeightRange{Start: fromHeight, End: fromHeight + p.cfg.BlockBatchSize - 1}:
```

Comment:
Any reason we are changing this ?? We are reducing the blocksize from what it previously used to be . Couldn't it be handled by checking if the difference is <2 (or equal to one) in the lower file and keep other logic as it is.

Reply:
this is to prevent duplicate blocks sync.

for example, when batch side was added it generates like {1,2}, {2,3}, {3,4} but this fixes and does {1,2}, {3, 4}, {5,6}.

Reply:
but in the lower query, we are only using startheight for equality and isn't that the expected result ?? can you write a testcase to make it more clear ?

Reply:
Can you give an example of what would be generated for case of equals query and another one for get and Lte query ?

Reply:
it will be simply incrementally by one from the start height.

Reply:
Examples for both case would be much better to understand why we are using <1 condition compared to <2(or =1) case for the check.

Reply:
example:

<1: {1,1} = true
<2: {1,2} = true

Reply:
I meant both cases where we use equal query and (gte and lte) query if you missed the above comment.

Reply:
As I said:

> <1: {1,1}

Is when batch block is set to 1, so `eq` is executed

> <2: {1,2}

with full batch support because 2 block to sync